### PR TITLE
Allow for AHU profiles of different lengths

### LINKED ISSUE #553: AHU profile for durations other than 1 day
## What is the problem?

For an office zone, I would like to specify different volume flow profiles for workdays and for the weekend. Just passing a list of length 7 * 24 to `building.central_ahu.profile_v_flow` does not work, because when `export_aixlib()` calls `modelica_AHU_boundary()`, it runs 

```Python
        if time_line is None:
            time_line = self.create_profile()
``` 

and `create_profile()` uses its default parameters `create_profile(duration_profile=86400, time_step=3600, double=False)`. 

## Why do we want to solve it?

- In my opinion, it would be useful to allow users to specify different AHU operation schedules for periods other than 1 day repeated for the entire year.

## How do we want to solve it?

- I am just playing around with different use_conditions, so it's likely that I am just approaching the problem from a wrong angle.
- If there is no other way already implemented to set AHU operation for workdays and weekends, I suggest to not use the default value of `duration_profile=86400` in `create_profile()` , but instead simply determine the length of  `building.central_ahu.profile_v_flow` and create a corresponding `time_line`.

## Next steps

- If someone (@PRemmen , @MichaMans ?) could provide some feedback whether this is indeed a problem or whether there is a different approach to solving my problem and a solution is needed, I could implement the suggested solution.

### PR DESCRIPTION
**Pull request by marcusfuchs**
Closes #553. Change the AHU profile processing to work with different lengths for the profiles, as long as the longest profile's length is a multiple of all the shorter profiles' lengths.

@DaJansenGit @nhuelsenbeck : This seems to be related to your work in #544, too, so it would be great if you could also take a look at this.

### CODE REVIEW

File: tests/test_data.py [hunk 1]
```diff
@@ -3159,3 +3159,71 @@ def test_modelica_export_version(self):
...
+                if day < 5:
+                    ratio = val
+                else:
+                    if val == 1:
+                        ratio = 0.2
+                    else:
+                        ratio = 0.0
+                v_flow_week.append(ratio)
+
+        for building in prj_test.buildings:
```
Comment (MichaMans) [inline tests/test_data.py]: Could we test some kind of original values with the values implemented by hand to be sure that the same things are happening?

**Comment by marcusfuchs** [comment]
@MichaMans Thanks for the feedback. I must admit that it has been a while since I worked on this, so I will have to re-familiarize myself with the issue. I think I also saw that somewhere else in the code there was a similar setup for profiles but handled separately and a bit differently. I'll try to have another look at the code and your comments. Yet, unfortunately I cannot make time for this immediately, so it may take a while for me to re-visit this.

**Comment by PRemmen** [comment]
@marcusfuchs I'll did something very similiar for heating, cooling and internal gains profiles. In my opinion my approach is a bit more simple and also better to maintain:

The workflow is as follows:

1. pass over a list of any length (e.g. one day, one week, one month) with the desired values
2. the setter will use this list to fill a dataframe in UseConditions for a whole year: https://github.com/RWTH-EBC/TEASER/blob/issue510_usecond/teaser/logic/buildingobjects/useconditions.py#L134
3. despite some definition of dataframe the generation for the output files (now all txt) are just a few lines https://github.com/RWTH-EBC/TEASER/blob/issue510_usecond/teaser/logic/buildingobjects/calculation/aixlib.py#L184 

I propose that I'll adapt that also to the AHU profiles. From what i see this should not "destroy" anything in your workflow.

**Comment by marcusfuchs** [comment]
@PRemmen Great suggestion. Back when I proposed this change, I needed a fast solution for my special application. I only later realized that there are many other parts of TEASER where a similar solution would help and that my solution does not cover all cases gracefully. Therefore I am perfectly happy with closing this PR and adapting to your solution.

**Comment by PRemmen** [comment]
This issue will be handeld in issue510_usecond #510

**Pull request closed without merging**